v2/spanner-to-sourcedb/src/main/java/com/google/cloud/teleport/v2/templates/transforms/ProcessInformationSchemaFn.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (C) 2025 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.cloud.teleport.v2.templates.transforms;
+
+import static com.google.cloud.teleport.v2.templates.transforms.SpannerInformationSchemaProcessorTransform.MAIN_DDL_TAG;
+import static com.google.cloud.teleport.v2.templates.transforms.SpannerInformationSchemaProcessorTransform.SHADOW_TABLE_DDL_TAG;
+
+import com.google.cloud.spanner.BatchClient;
+import com.google.cloud.spanner.BatchReadOnlyTransaction;
+import com.google.cloud.spanner.DatabaseAdminClient;
+import com.google.cloud.spanner.Dialect;
+import com.google.cloud.spanner.TimestampBound;
+import com.google.cloud.teleport.v2.spanner.ddl.Ddl;
+import com.google.cloud.teleport.v2.spanner.ddl.InformationSchemaScanner;
+import com.google.cloud.teleport.v2.templates.utils.ShadowTableCreator;
+import org.apache.beam.sdk.io.gcp.spanner.SpannerAccessor;
+import org.apache.beam.sdk.io.gcp.spanner.SpannerConfig;
+import org.apache.beam.sdk.transforms.DoFn;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ProcessInformationSchemaFn extends DoFn<Void, Ddl> {
```

Comment:
Looks like other `Fn` classes have unit tests added. Can you add unit test for this? 

Please also have a look at any other new classes you have created in this PR and add unit tests for them.